internal/pkg/service/appsproxy/proxy/apphandler/apphandler.go

```diff
@@ -127,6 +128,11 @@ func (h *appHandler) serveHTTPOrError(w http.ResponseWriter, req *http.Request)
 		}
 	}
 
+	// Add request ID header, for OAuth2Proxy internals and app itself
+	if id := middleware.RequestIDFromContext(req.Context()); id != "" { //nolint:contextcheck // false positive
+		req.Header.Set(config.RequestIDHeader, id)
+	}
+
```

Comment:
Update request, add request ID header, it is used by the OAuth2Proxy (so no new internal request id is generated), and it can be also used by the app itself.